dracut/10usr-generator/usr-generator

```diff
@@ -58,7 +59,19 @@ esac
 
 # Only proceed if the source is a path.
 if [[ "${usr}" != /* ]]; then
+  if [[ -z "${usr}" && -f /usr.squashfs ]]; then
+    # Don't set "norecovery" for squashfs
     exit 0
+  fi
+  # Add "norecovery" to systemd-fstab-generator's unit and exit
+  mount_usrflags=$(cmdline_arg mount.usrflags ro)
+  mount_usrflags="${mount_usrflags},norecovery"
```

Comment:
is it worth changing to add `norecovery` only if `ro` is specified? If an image is built with `--noenable_rootfs_verification` then `norecovery` is probably not a good idea.

Reply:
Yes, we could do it but I'm not really sure because it introduces a different behavior compared to ext2. With ext2 the filesystem is mounted read-only (but without dm-verity) and no changes happen until fsck is run, thus the image checksum stays correct. With btrfs I tried to preserve this behavior of not changing the image checksum implicitly, but I need to try out if it's actually too conservative and whether `recovery` only kicks in if the image was changed already anyway.

Reply:
Probably it's enough to test the behavior with a loopback device and run md5sum on it.

Reply:
I did as you suggested and added a check for `ro`